frappe/core/api/user_invitation.py

```diff
@@ -0,0 +1,110 @@
+import frappe
+
+USER_INVITATION_DOCTYPE = "User Invitation"
+FRAMEWORK_APP_NAME = " "
+
+
+@frappe.whitelist()
+def invite_by_email(
+	emails: str, role: str, redirect_to_path: str, app_name: str = FRAMEWORK_APP_NAME
+) -> dict[str, list[str]]:
+	frappe.only_for(["System Manager", "User Invitation Manager"])
+	is_app_framework = app_name == FRAMEWORK_APP_NAME
+	allowed_roles: list[str] = []
+	if not is_app_framework:
+		# `role` validation
+		invalid_app_msg = frappe._("Invalid app")
```

Comment:
1. Prefer importing `_` separately. Improves readability. 
2. Write better error message. Use different message for different failures. This doesn't explain anything. 
3. Add title to error message. 

https://gameplan.frappe.cloud/g/space/94/discussion/2114/gyan-write-better-error-messages